doc/proposals/scorecard-security.md

```diff
@@ -0,0 +1,87 @@
+
+## Release Signoff Checklist
+
+- \[ X\] Enhancement is `provisional`
+- \[ X\] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in [operator-sdk/doc][operator-sdk-doc]
+
+## Open Questions (optional)
+
+- Would have more security checks which may are important for the Operators and are not mapped here? 
+
+## Summary
+
+Currently, the scorecard will perform some tests. For further information see its [doc](https://github.com/operator-framework/operator-sdk/blob/master/doc/test-framework/scorecard.md#tests-performed).  
+This proposal is regarded to add a new suite of tests in order to ensure the quality of the operators by checking if they are complying with few common/general security practices. 
+In order to clarifies and exemplify this need as solution see the project https://kubesec.io/ and its demo which verifies and check if the resource is complying with the security common practices.   
+
+## Motivation
+
+- Ensure the quality of the operators projects by checking their security.   
+- Ensure the quality by checking the security of the projects which are distributed by OperatorHub.io and OpenShift catalogue
+
+## Goals
+
+* The test should fail when 1 or more resources created by the project are not complaining with the security rules adopted
+
+## Proposal
+
+### User Stories (optional)
+
+#### Story 1
+
+I as an operator dev user, I'd like to check the quality of my operator by verifying via tooling if it is complying with the minimal/common requirements. 
+
+**Acceptance Criteria**
+
+- The tooling/check should fail if the operator project creates a resource which not complain with the roles defined
+- The tooling/check should fail if the operator project do not label all resources created by it
+- The tooling/check should show what/where/why the operator did not pass in the tests for each failure rule
+
+### Implementation Details/Notes/Constraints
+
+For the following checks is possible to verify the Operator.YAML file:
+
+- Check if Operator has resources limits defined (memory/CPU)
+- Check if Operator has requests limits setup (memory/CPU)
+
+```yaml
+          resources:
+            limits:
+              cpu: 60m
+              memory: 128Mi
+            requests:
+              cpu: 30m
+              memory: 64Mi
+```
+
+For the following checks would be required to have the operator and its CR/CRD's applied in the cluster in order to get all pods/deployments created by it. 
+
+- Check if Pods/Deployments containers are using root as user
+- Check if Pods/Deployments has resources limits setup (memory/cpu)
+- Check if Pods/Deployments has requests limits setup (memory/cpu)
+- Check if Pods/Deployments containers are using privileged capability as SYS_ADMIN and CAP_SYS_ADMIN
+- Check if Pods/Deployments containers are using user ID > 10000
```

Comment:
I am interested in how this plays with OpenShifts default behavior where it does not assume a specific user ID being set but rather a random one will be picked every time.

Reply:
It is defined by the annotation `openshift.io/sa.scc.uid-range`. See [here](https://docs.openshift.com/enterprise/3.1/architecture/additional_concepts/authorization.html#understanding-pre-allocated-values-and-security-context-constraints)

Reply:
Info added in the proposal as well so I am closing it.